src/teleop/jetson/jetson_teleop.py

```diff
@@ -97,37 +100,108 @@ def __init__(self):
             "finger",
             "gripper",
         ]
+        SA_NAMES = ["joint_a", "joint_b", "joint_c", "joint_d", "scoop", "microscope"]
+
         self.ra_cmd = JointState(
-            name=[name for name in self.ra_names],
-            position=[math.nan for i in range(len(self.ra_names))],
-            velocity=[0.0 for i in range(len(self.ra_names))],
-            effort=[math.nan for i in range(len(self.ra_names))],
+            name=[name for name in RA_NAMES],
+            position=[math.nan for i in range(len(RA_NAMES))],
+            velocity=[0.0 for i in range(len(RA_NAMES))],
+            effort=[math.nan for i in range(len(RA_NAMES))],
+        )
+
+        self.sa_cmd = JointState(
+            name=[name for name in SA_NAMES],
+            position=[math.nan for i in range(len(SA_NAMES))],
+            velocity=[0.0 for i in range(len(SA_NAMES))],
+            effort=[math.nan for i in range(len(SA_NAMES))],
         )
 
-    def ra_control_callback(self, msg):
+    def filter_xbox_axis(
+        self,
+        axes_array: "list[float]",
+        axis_name: str,
+        deadzone_threshold: float,
+        quad_control: bool = True,
+    ) -> float:
+        """
+        Applies various filtering functions to an axis for controlling the arm
+        :param axes_array: Axis array from sensor_msgs/Joy, each value is a float from [-1,1]
+        :param axis_name: String representing the axis you are controlling, should match teleop.yaml
+        :param deadzone_threshold: Float representing the deadzone of the axis that you would like to use
+        :param quad_control: Bool for whether or not we want the axis to follow an x^2 curve instead of a linear one
+        Velocities are sent in range [-1,1]
+        :return: Returns an output velocity value for the given joint using the given axis_name
+        """
+        return (
+            quadratic(deadzone(axes_array[self.xbox_mappings[axis_name]], deadzone_threshold))
+            if quad_control
+            else deadzone(axes_array[self.xbox_mappings[axis_name]], deadzone_threshold)
+        )
```

Comment:
Don't see a reason for this code duplication, can just set the deadzone() to some variable and call quadratic if necessary

Reply:
I just thought putting it all in one function made sense, that way it's easier to switch out/add config params for in the future

Reply:
Yeah the function seems fine, just doesn't need to call deadzone in 2 different spots